store/mongo/migration_1_2_16.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type migration_1_2_16 struct {
+	client *mongo.Client
+	db     string
+}
+
+// Up creates an index for filtering tags and retrieving distinct tag keys.
+func (m *migration_1_2_16) Up(from migrate.Version) error {
+	ctx := context.Background()
+	idxReleases := m.client.
+		Database(m.db).
+		Collection(CollectionReleases).
+		Indexes()
+
+	_, err := idxReleases.CreateOne(ctx, mongo.IndexModel{
+		Keys: bson.D{{
+			Key:   StorageKeyReleaseTags,
```

Comment:
MongoDB calls array-type indexes for [multi-key indexes](https://www.mongodb.com/docs/manual/core/index-multikey/). An index on an array creates a node for each element in the array per document. All indexes in MongoDB are B+ trees, so the performance for lookup on a tag is `O(log(NT))`, where `N` is the number of documents in the collection and `T` is the number of tag elements per array (since the latter is constrained to maximum 100 tags, the complexity is `O(log N)`. You can read more about multi-key indexes in the [official documentation](https://www.mongodb.com/docs/manual/core/index-multikey/).

Reply:
thank you!